internal/api/download_test.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/opg-sirius-finance-admin/internal/model"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSubmitDownload(t *testing.T) {
+	mockClient := SetUpTest()
+	client, _ := NewApiClient(mockClient, "http://localhost:3000", "")
+
+	data := `{
+		"reportType":         "AccountsReceivable",
+		"reportJournalType":  "",
+		"reportScheduleType": "",
+		"reportAccountType":  "BadDebtWriteOffReport",
+		"reportDebtType":     "",
+		"dateOfTransaction":          "11/05/2024",
+		"dateFrom":      "01/04/2024",
+		"dateTo":        "31/03/2025",
+		"email":         "SomeSortOfEmail@example.com",
+	}
+	`
+
+	r := io.NopCloser(bytes.NewReader([]byte(data)))
+
+	GetDoFunc = func(*http.Request) (*http.Response, error) {
+		return &http.Response{
+			StatusCode: 201,
+			Body:       r,
+		}, nil
+	}
+
+	err := client.Download(getContext(nil), "AccountsReceivable", "", "", "BadDebtWriteOffReport", "", "11/05/2024", "01/04/2024", "31/03/2025", "SomeSortOfEmail@example.com")
+	assert.Equal(t, nil, err)
+}
+
+func TestSubmitDownloadUnauthorised(t *testing.T) {
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'd stick to one style of testing, either use the mock to switch behaviour _or_ do a test server

Reply:
done